src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -165,18 +165,19 @@ void PreReleaseRequest<I>::send_shut_down_image_cache() {
   CephContext *cct = m_image_ctx.cct;
   ldout(cct, 10) << dendl;
 
-  /* Shut down existing image cache whether the feature bit is on or not */
-  if (!m_image_ctx.image_cache) {
+  if (!m_image_ctx.io_image_dispatcher->dispatch_exist(
```

Comment:
Nit: I'd rather see `io::Dispatcher::shut_down_dispatch` be tweaked to just ignore missing dispatch layers and invoke the provided callback via the `op_work_queue`

Reply:
Removed dispatch_exist, and did update in shut_down_dispatch().